### DISCUSSION TITLE
Sequential patches with Queue

### DISCUSSION THREAD

**Discussion by soumickmj** [TorchIO-project/torchio#467] [Ideas]
Hi Fernando and everyone,
Is there any way by which I can get all the patches with certain overlap (similar to Grid sampler) with Patch Queue?
Or is there something like a "Queue of Grid Samplers" OR "Grid sampler of Subjects"?
My motivation is to be able to use during validate stage, so I'm not gonna be using Grid aggregator. But I need to be able to sample all the patches, and not randomly. 
Any suggestions?
Thanks.

**Comment by soumickmj** [comment]
Btw, I'm currently using Concat dataset to do the job.

```python
            grid_samplers = []
            for i in range(len(subjects)):
                grid_sampler = tio.inference.GridSampler(
                                                            subjects[i],
                                                            self.patch_size,
                                                            (4,4,4),
                                                        )
                grid_samplers.append(grid_sampler)
            return torch.utils.data.ConcatDataset(grid_samplers)
```

But I was wondering maybe there is already some more elegant way of doing this?

**Reply by fepegar** [reply]
Hi, @soumickmj. Why not just going over each subject in a validation dataset and creating a new grid sampler for each of them, from which you can extract patches for validation?
